gwctl/pkg/printer/gatewayclasses_test.go

```diff
@@ -281,18 +282,11 @@ func TestGatewayClassesPrinter_LabelSelector(t *testing.T) {
 		t.Fatalf("Failed to construct resourceModel: %v", resourceModel)
 	}
 
-	gcp := &GatewayClassesPrinter{
-		Out:   params.Out,
-		Clock: fakeClock,
+	expectedGatewayClassNames := []string{"foo-com-internal-gateway-class"}
+	gatewayClassNames := make([]string, 0, len(resourceModel.GatewayClasses))
+	for _, gatewayClassNode := range resourceModel.GatewayClasses {
+		gatewayClassNames = append(gatewayClassNames, gatewayClassNode.GatewayClass.GetName())
 	}
-	gcp.Print(resourceModel)
 
-	got := params.Out.(*bytes.Buffer).String()
-	want := `
-NAME                            CONTROLLER                                 ACCEPTED  AGE
-foo-com-internal-gateway-class  foo-com-internal-gateway-class/controller  True      365d
-`
-	if diff := cmp.Diff(common.YamlString(want), common.YamlString(got), common.YamlStringTransformer); diff != "" {
-		t.Errorf("Unexpected diff\ngot=\n%v\nwant=\n%v\ndiff (-want +got)=\n%v", got, want, diff)
-	}
+	assert.Equal(t, expectedGatewayClassNames, gatewayClassNames, "Expected GatewayClass name does not match the found one")
```

Comment:
Non blocking: Let's try to avoid a new dependency if possible. Instead, let's not shy away from writing something like: 

```
	if diff := cmp.Diff(expectedGatewayClassNames, gatewayClassNames); diff != "" {
		t.Errorf("DiscoverResourcesForGatewayClass(%+v) returned unexpected diff (-want, +got):\n%v", filter, diff)
	}
```

(Personally, I [try to avoid assertion libraries](https://google.github.io/styleguide/go/decisions#assertion-libraries) but wiser people than me do infact use it so can't really complain and don't want to enforce)

Reply:
Yup, done!